### PR TITLE
feat: add snapshotMaxCount and snapshotMaxSize to Volume API

### PR DESCRIPTION
**Pull request by FrankYang0529**
#### Which issue(s) this PR fixes:

https://github.com/longhorn/longhorn/issues/7522

#### What this PR does / why we need it:

Support `snapshotMaxCount` and `snapshotMaxSize` in LH Volume API.

### CODE REVIEW

**Review by derekbit — DISMISSED** [review]
LGTM

File: api/volume.go [hunk 1]
```diff
@@ -167,6 +167,11 @@ func (s *Server) VolumeCreate(rw http.ResponseWriter, req *http.Request) error {
 		}
 	}
 
+	snapshotMaxSize, err := util.ConvertSize(volume.SnapshotMaxSize)
+	if err != nil {
+		return fmt.Errorf("failed to parse snapshot max size %v", err)
```
Comment (derekbit) [inline api/volume.go:172]: nit: 
```suggestion
		return errors.Wrap(err, "failed to parse snapshot max size")
```

File: api/model.go [hunk 2]
```diff
@@ -60,6 +60,8 @@ type Volume struct {
 	DataEngine                       longhorn.DataEngineType                `json:"dataEngine"`
 	OfflineReplicaRebuilding         longhorn.OfflineReplicaRebuilding      `json:"offlineReplicaRebuilding"`
 	OfflineReplicaRebuildingRequired bool                                   `json:"offlineReplicaRebuildingRequired"`
+	SnapshotMaxCount                 int                                    `json:"snapshotMaxCount"`
+	SnapshotMaxSize                  string                                 `json:"snapshotMaxSize"`
```
Comment (innobead) [inline api/model.go:64]: Why the type is string?

**Review by innobead — COMMENTED** [review]
Just one question.

File: api/volume.go (hunk 1 above)
Comment (FrankYang0529) [reply api/volume.go]: Updated it. Thanks.

File: api/model.go (hunk 2 above)
Comment (FrankYang0529) [reply api/model.go:64]: It's like `Size`. We use string for it.

https://github.com/longhorn/longhorn-manager/blob/591946024a333421dc446afeea62618525fdb42a/api/model.go#L33

Comment (innobead) [reply api/model.go:64]: I see. Should revisit why we did that way later.

**Review by innobead — APPROVED** [review]

**Comment by FrankYang0529** [comment]
> Should this also return the actions for updating the values? It seems that they are currently missing

Yeah, let me add it. Thanks.

**Pull request merged**